### DISCUSSION TITLE
Question on relay host relationship with DKIM

### DISCUSSION THREAD

**Discussion by salderma** [docker-mailserver/docker-mailserver#2656] [Q&A]
Hi, I'm looking at using sendgrid, been operating docker-mailserver for a few years w/o issues, but all of a sudden my ISP's ASN has been added to a blacklist, they won't help and neither will the blacklist.

Anyway, I thought I'd try sendgrid, I've used Twilio's other offerings. Has anyone been through their setup with this project? I'm mostly curious about their automatic DKIM security and it's relationship with the DKIM setup I already have using this project.  Secondly, if anyone with experience has some advice on the remaining settings Sendgrid that I haven't reached passed the "Advanced Settings" on Domain Authentication.

I realize these trusted sender services change their offerings and configurations from time to time, so it's not exactly the domain of this project to have a setup tutorial. Just trying to avoid screwing things up or making bad choices in the setup on either side.

Any advice?

**Comment by ferdinandyb** [comment]
Did you go through with your setup? I also need to use a relay which already has dkim, in the meanwhile postfix is complaining about opendkim not running, but my hunch is that it's not a good idea to add dkim and then send to relay?

**Reply by salderma** [reply]
No, while my ISP's subnet still remains on the blacklist, I have not had many delivery problems since trying to look into this 6 months ago.

**Reply by ferdinandyb** [reply]
thanks!

**Reply by ferdinandyb** [reply]
in the end I edited the `SigningTable` file and disabled any actual signing because some places didn't like the double DKIM and were sending back failed dmarc messages.
